### ISSUE TITLE
Sway stops responding to idle command from swayidle

### ISSUE THREAD

**Issue by pielgrzym** [swaywm/sway#7545]
- **Sway Version:**
  - 1.8.1

- **Debug Log:**
  - https://gist.github.com/pielgrzym/bd3a99264d86e0fe52d5058353af003e

I'm using swayidle with such invocation:
```sh
swayidle -w timeout 600 'swaylock' timeout 1200 'swaymsg "output * dpms off"' resume 'swaymsg "output * dpms on"' before-sleep 'swaylock'
```
I use `killall -USR1 swayidle` to trigger screen lock.
When I have external monitor attached for longer period of time the USR1 signal does not lock the screen or locks the screen only after I invoke a keybinding to change window. In the log in gist there is a case when the screen actually locked but with 10 minutes delay. When I detach monitor screen locking starts working again. Sometimes reloading sway does the trick without detaching monitor. Other times it just starts working again. It looks like the event produced by swayidle uppon USR1 is consumed by sway with a large delay - perhaps there is some queue that is filled up?

Is there a workaround this problem? This seriously undermines system security when your automatic screen locker works from time to time. Perhaps there is an alternative way I can detect idle and lock screen?

**Comment by joanbm** [comment]
Hmmmm, not sure about the problem with `SIGUSR1`, but as a workaround, maybe you can lock the screen by running `swaylock` directly, instead of `killall -USR1 swayidle`?

(Note that only a single instance of `swaylock` can run, so the screen won't be "double locked" after 10 minutes).

**Comment by pielgrzym** [comment]
Thanks, I actually used this fallback :) swayidle is triggering DPMS after longer inactivity (or not - it's a hit or miss, sometimes it does, othertimes it doesn't).

**Comment by joanbm** [comment]
Yeah, not sure if it's related in some way to your issue, but swayidle's handling of `SIGUSR1` has some warts like https://github.com/swaywm/swayidle/issues/51 that makes it unusable as a "I'm going away, run all the inactivity triggers right now" button.

**Comment by pielgrzym** [comment]
I tried reporting the issue there, but they  said it's sway's fault :( I did some tests and for sure the SIGUSR1 is received (debug logs indicate this), however the events are not fired. I looked into swayidle code and there is almost no logic there - everything is handled by [ext-idle-notify](https://gitlab.freedesktop.org/wayland/wayland-protocols/-/tree/main/staging/ext-idle-notify) so I guess this is the faulty part, however I have no idea how to debug this so I can provied an actionable issue. This behavior is so frustrating when you can't know for sure if your screen will lock automatically or if it will shut down for the night :(

**Comment by wiktor-k** [comment]
> or locks the screen only after I invoke a keybinding to change window.

Huh, I thought I was going crazy (`swaylock` gets the signal but apparently nothing happens) until I read this and used `$mod+2`. Lo and behold - swaylock started working and correctly turned off the screen. Really interesting behavior as I haven't seen that previously (I'm on Arch and everything is up-to-date).
